core/src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java

```diff
@@ -418,25 +422,35 @@ public static long nodeLongValue(Object node) {
     /**
      * This method is very lenient, use {@link #nodeBooleanValue} instead.
      */
-    public static boolean lenientNodeBooleanValue(Object node, boolean defaultValue) {
+    public static boolean lenientNodeBooleanValue(Object node, String name, boolean defaultValue) {
         if (node == null) {
             return defaultValue;
         }
-        return lenientNodeBooleanValue(node);
+        return lenientNodeBooleanValue(node, name);
     }
 
     /**
      * This method is very lenient, use {@link #nodeBooleanValue} instead.
      */
-    public static boolean lenientNodeBooleanValue(Object node) {
+    public static boolean lenientNodeBooleanValue(Object node, String name) {
+        boolean interpretedAsLenient = false;
+        boolean booleanValue;
+
         if (node instanceof Boolean) {
-            return (Boolean) node;
+            booleanValue = (Boolean) node;
+        } else if (node instanceof Number) {
+            interpretedAsLenient = true;
+            booleanValue = ((Number) node).intValue() != 0;
+        } else {
+            String value = node.toString();
+            booleanValue = !(value.equals("false") || value.equals("0") || value.equals("off"));
```

Comment:
Hmm I see the original behavior had this inconsistency also (not parsing `"no"` like all the other boolean methods do)... not sure whether we should try to re-add the consistency or keep the broken behavior.

I'm in favor of re-adding the `"no"` parsing to be consistent as a semi-bugfix, what do you think? (it's going away in 6.0 anyway...)